crates/oxc_hir/src/hir.rs

```diff
@@ -123,7 +123,13 @@ impl<'a> Expression<'a> {
 
     /// Determines whether the given expr is a `void 0`
     pub fn is_void_0(&self) -> bool {
-        matches!(self, Self::UnaryExpression(expr) if expr.operator == UnaryOperator::Void)
+        if let Self::UnaryExpression(expr) = self
+            && expr.operator == UnaryOperator::Void
+            && let Self::NumberLiteral(lit) = &expr.argument
+            && lit.value == 0.0 {
+            return true
+        }
+        false
```

Comment:
Can we reduce the code duplication in `is_void_0` in multiple crates?

Reply:
Unfortunately not at the moment, they are different structs. I haven't found a way to shared code between the AST and the HIR.